### PR TITLE
Add window method tests

### PR DESCRIPTION
**Pull request by AndyMJones**

### CODE REVIEW

File: packages/api-tests/test/window.spec.js [hunk 1]
```diff
@@ -54,16 +54,328 @@ describe('Window API', function(done) {
...
+        assert.equal(result.value.width, width)
+        assert.equal(result.value.x, x)
+        assert.equal(result.value.y, y)
+      }
+    ];
+
+    return chainPromises(steps);
+  });
+
+  it('Should get if the window is alwaysOnTop', function() {
```
Comment (BenLambertNcl) [inline packages/api-tests/test/window.spec.js]: Guessing there's a word missing here?

File: packages/api-tests/test/window.spec.js [hunk 2]
```diff
@@ -54,16 +54,328 @@ describe('Window API', function(done) {
...
+      () => selectWindow(app.client, 1),
+      () => app.client.waitForVisible('.visible-check'),
+      () => callWindowMethod('isAlwaysOnTop'),
+      (result) => assert.equal(result.value, alwaysOnTop)
+    ];
+
+    return chainPromises(steps);
+  });
+
+  it('Should get if the window is maximizable', function() {
```
Comment (BenLambertNcl) [inline packages/api-tests/test/window.spec.js]: ditto. There are a few more too further down

File: packages/api-tests/test/window.spec.js [hunk 3]
```diff
@@ -54,16 +54,328 @@ describe('Window API', function(done) {
     return executeAsyncJavascript(app.client, script);
   });
 
+  it('Should close the window', function() {
```
Comment (BenLambertNcl) [inline packages/api-tests/test/window.spec.js]: Think having all of these in a describe would be nice so we can group the window option tests together, the window method tests together etc

File: packages/api-tests/test/window.spec.js [hunk 4]
```diff
@@ -54,16 +54,328 @@ describe('Window API', function(done) {
...
+    const windowTitle = 'windownameclose';
+    const windowOptions = {
```
Comment (BenLambertNcl) [inline packages/api-tests/test/window.spec.js]: Could have a global `defaultWindowOptions` and just override what we need for each test?

**Review by BenLambertNcl — COMMENTED** [review]
Few comments, but looking good :+1:

**Review by BenLambertNcl — APPROVED** [review]

**Review by ColinEberhardt — COMMENTED** [review]
There's quite a bit of repeated code in the tests. Could you create a utility function that constructs a window ready for testing?

```
const createWindow(windowTitle) => {
      const windowOptions = getWindowOptions({
        name: windowTitle
      });

      const steps = [
        () => app.client.isVisible('.visible-check'),
        () => openNewWindow(app.client, windowOptions),
        () => selectWindow(app.client, 1),
        () => app.client.waitForVisible('.visible-check')
    ]
    return chainPromises(steps);
}
```

**Pull request merged**